src/plugins/persistence/couch/CouchChangesFeed.js

```diff
@@ -13,17 +13,19 @@
             connectionId: connections.length
         });
 
-        port.onmessage = async function (event) {
+        port.onmessage = function (event) {
             if (event.data.request === 'close') {
-                console.log('Closing connection');
+                console.debug('🚪 Closing couch connection 🚪');
                 connections.splice(event.data.connectionId - 1, 1);
                 if (connections.length <= 0) {
                     // abort any outstanding requests if there's nobody listening to it.
                     controller.abort();
                 }
 
-                console.log('Closed.');
                 connected = false;
+                // stop listening for events
+                couchEventSource.removeEventListener('message', self.sourceListener);
```

Comment:
I'm trying to map this to a corresponding `addEventListener` call, and failing. 

The `addEventListener('message'` paradigm is much preferable to `.onmessage=`, so I'm 👍 on this approach, I just don't see a corresponding call setting up this listener.